doc/source/cluster/slurm.rst

```diff
@@ -92,3 +92,225 @@ Starter SLURM script
       end = time.time()
       print(end - start)
+
+
+Fully functional SLURM scripts
+------------------------------
+
+The starter SBATCH script does the following things:
+
+1. It fetches the list of computing nodes and their IP addresses.
+2. It launches a head ray process in one of the node (called the head node), and get the address of the head node.
+3. It launches ray processes in (n-1) worker nodes and connects them to the head node by providing the head node address.
+4. After the underlying ray cluster is ready, it submits the user specified task.
+
+Extending the starter script above, we can formalize a fully functional SLURM scripts with helpful python interface.
+First, we rewrite the starter SBATCH script and add some placeholders so that we can use a python script to fulfill them.
+Then, we can write a python interface to take user's specification as argument and replace the placeholders in the SBATCH scripts.
+The python script can also be used to launch the Ray job to the ray cluster which is hosted inside the slurm cluster.
+
+The following `launch.py` and `sbatch_template.sh` implement this idea and you can copy them to your own project as helper files.
+Concretely, the `launch.py` does the following things:
+
+1. It automatically writes your requirements, e.g. number of CPUs, GPUs per node, the number of nodes and so on, to a sbatch script name `{exp-name}_{date}-{time}.sh`. Your command (`--command`) to launch your own job is also written into the sbatch script.
+2. Then it will submit the sbatch script to slurm manager via a new process.
+3. Finally, the python process will terminate itself and leaves a log file named `{exp-name}_{date}-{time}.log` to record the progress of your submitted command. At the mean time, the ray cluster and your job is running in the slurm cluster.
+
+
+If you want to utilize multiple computing node in slurm and let ray recognizes them, please use:
+
+.. code-block:: bash
+    python launch.py --exp-name test --command "python your_file.py" --num-nodes 3
+
+
+If you want to specify the computing nodes, just use the same node name in `sinfo` command:
+
+.. code-block:: bash
+    python launch.py --exp-name test --command "python your_file.py" --num-nodes 3 --node chpc-cn[003-005]
+
+
+There are other options you can use when calling `python launch.py`:
+
+* `--exp-name`: The experiment name. Will generate `{exp-name}_{date}-{time}.sh` and  `{exp-name}_{date}-{time}.log`.
+* `--command`: The command you wish to run. For example: `rllib train XXX` or `python XXX.py`.
+* `--num-gpus`: The number of GPUs you wish to use in each computing node. Default: 0.
+* `--node` (`-w`): The specify nodes you wish to use, in the same form of the return of `sinfo`. Automatically assign if not specify.
+* `--num-nodes` (`-n`): The number of nodes you wish to use. Default: 1.
+* `--partition` (`-p`): The partition you wish to use. Default: "chpc" (CUHK cluster partition name, change to yours!)
```

Comment:
Is chpc a common convention for naming slurm partitions? If not, maybe we should remove the default and make this a required flag

Reply:
Now I have change the default partition to "", in which case the slurm will look for the default partition for the user.